src/common/transformations/src/transformations/symbolic_transformations/dereshape_matmul.cpp

```diff
@@ -340,65 +341,38 @@ ov::pass::DeReshapeMatMul::DeReshapeMatMul() {
 
 ov::pass::DeReshapeFullyConnected::DeReshapeFullyConnected() {
     MATCHER_SCOPE(DeReshapeFullyConnected);
+    using namespace ov::op;
+    using namespace ov::pass::pattern;
 
-    auto reshaped_input = pattern::wrap_type<op::v1::Reshape>([](Output<Node> out) -> bool {
-        const auto& input_shape = out.get_node_shared_ptr()->get_input_partial_shape(0);
-        if (input_shape.rank().is_dynamic() || input_shape.size() < 3)
-            return false;
-        const auto& output_shape = out.get_partial_shape();
-        if (output_shape.rank().is_dynamic() || output_shape.size() < 2)
-            return false;
-        return dims_are_equal(input_shape[input_shape.size() - 1], output_shape[output_shape.size() - 1]);
-    });
-    auto converted =
-        pattern::wrap_type<op::v0::Convert>({reshaped_input}, pattern::consumers_count(1));  // optional convert
-
-    auto dynamic_input = std::make_shared<pattern::op::Or>(OutputVector{reshaped_input, converted});
-    auto static_input = pattern::any_input(pattern::rank_equals(2));
-    auto mm_label = pattern::wrap_type<op::v0::MatMul>({dynamic_input, static_input}, [](Output<Node> out) -> bool {
-        auto mm = ov::as_type_ptr<op::v0::MatMul>(out.get_node_shared_ptr());
-        return mm && !mm->get_transpose_a() && pattern::consumers_count(1)(out);
-    });
+    auto transpose_a_false = [](const std::shared_ptr<Node>& node) -> bool {
+        auto mm = as_type_ptr<v0::MatMul>(node);
+        return mm && !mm->get_transpose_a();
+    };
 
-    auto reshaped_output =
-        pattern::wrap_type<op::v1::Reshape>({mm_label, pattern::any_input()}, [](Output<Node> out) -> bool {
-            const auto& input_shape = out.get_node_shared_ptr()->get_input_partial_shape(0);
-            if (input_shape.rank().is_dynamic() || input_shape.size() < 2)
-                return false;
-            const auto& output_shape = out.get_partial_shape();
-            if (output_shape.rank().is_dynamic() || output_shape.size() < 3)
-                return false;
-            return dims_are_equal(input_shape[input_shape.size() - 1], output_shape[output_shape.size() - 1]);
-        });
+    auto input = wrap_type<v1::Reshape>({any_input(shape_matches("BATCHES_1...,Y")), any_input()},
+                                        shape_matches("BATCHES_2...,Y"));
+    auto converted = optional<v0::Convert>(input, consumers_count(1));
```

Comment:
```suggestion
    auto converted = optional<v0::Convert>(input, consumers_count(1));
```
optional add namespace prefix what kind of optimal it is